build/xtask/src/dist.rs

```diff
@@ -1064,6 +1075,272 @@ fn build_task(cfg: &PackageConfig, name: &str) -> Result<()> {
         .context(format!("failed to build {}", name))
 }
 
+fn task_can_overflow(
```

Comment:
Are there any concerns about false positives / false negatives? I guess for a false positive we'd increase the stack size unnecessarily and for a false negative, well, we're back to our old methods.

Reply:
Added more comments in 9471544fbf62e939a01757b91d3d2f6fd868e738 about the possibility of false positives (possible but unlikely) and negatives (possible if using dynamic dispatch).